examples/seismic/acoustic/wavesolver.py

```diff
@@ -102,15 +111,21 @@ def forward(self, src=None, rec=None, u=None, m=None, save=False, **kwargs):
                          space_order=self.space_order,
                          dtype=self.model.dtype)
 
+        kwargs.update({'u': u})
+
         # Pick m from model unless explicitly provided
-        if m is None:
-            m = m or self.model.m
+        if m is not None:
```

Comment:
And, most importantly: why not simply letting Devito fails if one tries to pass in a DenseData when ``m`` was instead set to a constant in the problem specification? In such a case, where would Devito die? @mlange05 any thoughts ?